src/test/java/com/example/rets_api/repository/PropertyRepositoryJPATest.java

```diff
@@ -138,6 +141,30 @@ public void should_update_a_Property() {
         comparePropertyPatchDTOBasicFields(PropertyConverter.propertyEntityToPropertyPatchDTO.convert(responseEntity), PropertyConverter.propertyEntityToPropertyPatchDTO.convert(propertyToPatch));
     }
 
+    @Test
+    public void should_update_a_Property_updateViewData() {
+        //Creating
+        ViewDataDTO viewDataDTOToPatch = createViewDataDTO();
+        ViewDataEntity viewDataEntityToPatch = ViewDataConverter.viewDataDTOToViewDataEntity.convert(viewDataDTOToPatch);
+        PropertyEntity propertyToPatch = EntityUtilsTest.createPropertyEntityWithDifferentBasicFields();
+        propertyToPatch.setViewData(viewDataEntityToPatch);
+        PropertyEntity propertyEntity = EntityUtilsTest.createPropertyEntityWithBasicFields();
+
+        //Saving
+        propertyEntity = propertyRepository.saveAndFlush(propertyEntity);
+
+        //Updating
+        PropertyEntity propertyToSave = updatePropertyValues(propertyEntity, propertyToPatch);
+
+        //Patching
+        PropertyEntity responseEntity = propertyRepository.saveAndFlush(propertyToSave);
+
+        //Validating
+        checkAllBasicFieldsFromProperty(responseEntity);
+        comparePropertyPatchDTOBasicFields(PropertyConverter.propertyEntityToPropertyPatchDTO.convert(responseEntity), PropertyConverter.propertyEntityToPropertyPatchDTO.convert(propertyToPatch));
```

Comment:
here comparePropertyPatchDTOBasicFields also compares only the basic fields and doesn't check the viewData 
I think we need a new method

Reply:
done :)